cheddar/cheddar-integration-aws/src/test/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDocumentStoreTemplateTest.java

```diff
@@ -175,6 +178,42 @@ public void shouldQueryIndex() {
     }
 
     @Test
+    public void shouldQueryIndex_withCompoundAttributeQuery() {
+        // Given
+        final ItemId itemId = new ItemId(randomId());
+
+        final ItemConfiguration itemConfiguration = new ItemConfiguration(StubWithGlobalSecondaryIndexItem.class,
+                tableName);
+        itemConfiguration.registerIndexes((Arrays.asList(new CompoundIndexDefinition("gsi", "gsiSupportingValue"))));
+        final Collection<ItemConfiguration> itemConfigurations = Arrays.asList(itemConfiguration);
+        when(mockDatabaseSchemaHolder.itemConfigurations()).thenReturn(itemConfigurations);
+
+        final Table mockTable = mock(Table.class);
+        when(mockDynamoDBClient.getTable(any(String.class))).thenReturn(mockTable);
+
+        final DynamoDocumentStoreTemplate dynamoDocumentStoreTemplate = new DynamoDocumentStoreTemplate(
+                mockDatabaseSchemaHolder);
+        dynamoDocumentStoreTemplate.initialize(mockAmazonDynamoDbClient);
+
+        final Index mockIndex = mock(Index.class);
+        when(mockTable.getIndex(anyString())).thenReturn(mockIndex);
+
+        final ItemCollection<QueryOutcome> outcome = mock(ItemCollection.class);
+        when(mockIndex.query(any(QuerySpec.class))).thenReturn(outcome);
+
+        // When
+        dynamoDocumentStoreTemplate.fetch(
```

Comment:
I followed the format of the existing tests which don't inspect that.  I've had a look at doing it and it would be a pain to do.  It involves mocking underlying Amazon classes which prove tricky to mock in random way (it stems from the `ItemCollection` class which uses its own implementation of an iterator).

The integration tests for this class test that the actual data returned from the fetch so this unit test class was simply testing that the correct paths were being followed.

I have added some inspection of what is returned from `fetch` but it is quite limited in that it is not random because of the complexities involved in mocking the required objects.